build.gradle

```diff
@@ -215,30 +215,6 @@ tasks.register("verifyVersions") {
   }
 }
 
-/*
- * When adding backcompat behavior that spans major versions, temporarily
- * disabling the backcompat tests is necessary. This flag controls
- * the enabled state of every bwc task. It should be set back to true
- * after the backport of the backcompat code is complete.
- */
-
-boolean bwc_tests_enabled = true
-// place a PR link here when committing bwc changes:
-String bwc_tests_disabled_issue = ""
-if (bwc_tests_enabled == false) {
-  if (bwc_tests_disabled_issue.isEmpty()) {
-    throw new GradleException("bwc_tests_disabled_issue must be set when bwc_tests_enabled == false")
-  }
-  println "========================= WARNING ========================="
-  println "         Backwards compatibility tests are disabled!"
-  println "See ${bwc_tests_disabled_issue}"
-  println "==========================================================="
-}
-if (project.gradle.startParameter.taskNames.any { it.startsWith("checkPart") || it == 'functionalTests' }) {
```

Comment:
Unfortunately we need to keep this bit or introduce a new toggle to skip BWC tests when we just want to run "functional" tests.

Reply:
Using this for `functionalTests` seems like a hack. Could we more explicitly define it? So instead of it being defined as depending on `check`, and filtering with this special flag and onlyIf, could we say "functional tests are x, y and z" tasks, and depend on those?